### PR TITLE
Trigger build if artifact is not found in transparency log

### PR DESCRIPTION
**Pull request by jperedadnr**
## Description

Fixes pyrsia/pyrsia#1101

This PR triggers a build in the case that an artifact requested is not found in the transparency log.

To prevent the same artifact is being built twice if two simultaneous request are done (via HEAD and GET for Docker), 
the fetch manifest has been split into two methods, one that checks if the artifact is present or not, but doesn't trigger a build, and one that checks if the artifact is present or not and triggers a build in case it is not.

The build request is async, and therefore the user that request a missing artifact will still get a non found artifact error. Subsequent requests to that artifact will work for this or any other user once the build is done. In case of Docker, the fallback to download from DockerHub is still in place, so the first user won't have a bad experience.

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/get_involved/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

- [x] I've built the code `cargo build --all-targets` successfully.
- [x] I've run the unit tests `cargo test --workspace` and everything passes.
- [x] I've made sure my rust toolchain is current `rustup update`.

### CODE REVIEW

**Review by sbtaylor15 — APPROVED** [review]
/lgtm

File: src/java/maven2/handlers/maven_artifacts.rs [hunk 1]
```diff
@@ -68,15 +95,12 @@ fn get_package_specific_artifact_id(full_path: &str) -> Result<String, anyhow::E
     if pieces.len() < 4 {
         bail!(format!("Error, invalid full path: {}", full_path));
     }
-    let file_name = pieces.pop().unwrap();
-    let version = pieces.pop().unwrap();
-    let artifact_id = pieces.pop().unwrap();
+    let file_name = pieces.pop().unwrap().to_string();
```
Comment (fishseabowl) [inline src/java/maven2/handlers/maven_artifacts.rs:98]: I'm worried that if something goes wrong, the main process will stop.

File: src/java/maven2/handlers/maven_artifacts.rs [hunk 2]
```diff
@@ -39,11 +46,16 @@ pub async fn handle_get_maven_artifact(
 
     // request artifact
     debug!(
-        "Requesting artifact for id {}",
-        package_specific_artifact_id
+        "Requesting artifact for id {} and artifact id {}. If not found a build will be requested",
```
[the remark this answers is not in the document]
Comment (jperedadnr) [reply src/java/maven2/handlers/maven_artifacts.rs:49]: Right, I'll add a more detailed message.

File: src/java/maven2/handlers/maven_artifacts.rs (hunk 1 above)
Comment (jperedadnr) [reply src/java/maven2/handlers/maven_artifacts.rs:98]: Okay, I'll add some guards here.

**Review by fishseabowl — APPROVED** [review]

**Review by efrisch — APPROVED** [review]
👍

**Pull request merged**